zegami_sdk/collection.py

```diff
@@ -1020,6 +1025,35 @@ def __len__(self) -> int:
     def __repr__(self) -> str:
         return "<Collection id={} name={}>".format(self.id, self.name)
 
+    def id_to_class(self, ID):
+        return self.classes[ID - 1]['name']
+
+    def get_annotations_metadata_to_csv(self, outpath='output.csv',
+                                        anno_type=None) -> pd.DataFrame:
+        """Creates a .csv file with annotations metadata from a collection.
+        Defaults to searching for annotations of all types.
+        """
+        metadata = self.get_annotations_metadata(anno_type)
+        metadata.to_csv(outpath)
+        print('Saved annotations metadata at "{}".'.format(outpath))
+        return metadata
+
+    def get_annotations_metadata(self, anno_type=None) -> pd.DataFrame:
+        """Compute pandas.DataFrame of annotations metadata from a collection.
+        Defaults to searching for annotations of all types.
+        """
+        annos = self.get_annotations(anno_type=anno_type)['annotations']
+        metadata = pd.DataFrame()
+        for i, anno in enumerate(annos):
+            row = {**{'Image': anno['image_index'],
+                      'ID': anno['id'],
+                      'Class': self.id_to_class(int(anno['class_id']))},
+                   **anno['metadata']}
+            metadata = metadata.append(row, ignore_index=True)
+            # Change order of columns
+        metadata = metadata[[metadata.columns[i] for i in (3, 2, 1, 0, 4, 5, 6)]]
```

Comment:
I agree that having the 'magic numbers' here doesn't seem quite right and won't work if the metadata isn't a specific length.
Better IMO to leave any ordering requirements to the 'client' script which receives this dataframe back.